src/main/java/application/onlinebookstore/repository/impl/BookRepositoryImpl.java

```diff
@@ -41,6 +42,26 @@ public Book save(Book book) {
         }
     }
 
+    @Override
+    public Optional<Book> findByTitle(String title) {
+        try (Session session = sessionFactory.openSession()) {
+            return Optional.of(session.get(Book.class, title));

```

Comment:
`Optional.of` throws NPE, check the docs. It's not what you want. Give the opportunity to decide whether to throw or not to the user of this class.